File: js/task2/S2_FOR_06.js
// S2_FOR_06 — Nested arrays

function sumNested(matrix) {
  let total = 0;

  for (const row of matrix) {
    for (const n of row) {
      total += n;
    }
  }

  return total;
}

// Tests
console.log(sumNested([[1, 2], [3, 4]]));        // 10
console.log(sumNested([[5], [10], [15]]));      // 30
console.log(sumNested([]));                     // 0
console.log(sumNested([[1, -1], [2, -2]]));     // 0

File: python/task2b/S2_FOR_06.py
# S2_FOR_06 — Sum nested lists (matrix)

def sum_nested(matrix):
    total = 0

    for row in matrix:
        if not isinstance(row, list):
            return None
        for value in row:
            total += value

    return total


def main():
    tests = [
        [[1, 2], [3, 4]],          # 10
        [[-1, 2, 3], [4]],         # 8
        [[]],                      # 0
        [],                        # 0
        [[1, 2], "not a list"],    # None
    ]

    for t in tests:
        print(f"matrix={t} -> sum={sum_nested(t)}")


if __name__ == "__main__":
    main()
